src/models/delegations.ts

```diff
@@ -69,13 +88,24 @@ export class Delegations {
 
     if (this.roles) {
       json.roles = rolesToJSON(this.roles);
+    } else if (this.succinctRoles) {
+      json.succinct_roles = this.succinctRoles.toJSON();
     }
 
     return json;
   }
 
   public static fromJSON(data: JSONObject): Delegations {
-    const { keys, roles, ...unrecognizedFields } = data;
+    const { keys, roles, succinct_roles, ...unrecognizedFields } = data;
+
+    if (isObject(succinct_roles)) {
+      return new Delegations({
+        keys: keysFromJSON(keys),
+        roles: rolesFromJSON(roles),
+        succinctRoles: SuccinctRoles.fromJSON(succinct_roles),
+        unrecognizedFields,
+      });
+    }
```

Comment:
Can we combine this with the code below somehow? Seems like a lot of it is duplicated.